appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/TransferAmountPresenter.kt

```diff
@@ -0,0 +1,197 @@
+package org.kinecosystem.appstransfer.presenter
+
+import android.os.Handler
+import org.kinecosystem.appstransfer.view.ITransferAmountView
+import org.kinecosystem.common.base.BasePresenter
+import org.kinecosystem.common.base.Consts
+import org.kinecosystem.common.base.Consts.TRANSACTION_TIMEOUT
+import org.kinecosystem.transfer.model.EcosystemApp
+import org.kinecosystem.transfer.model.getTransactionMemo
+import org.kinecosystem.transfer.model.iconUrl
+import org.kinecosystem.transfer.model.name
+import org.kinecosystem.transfer.repositories.EcosystemAppsRepository
+import org.kinecosystem.transfer.sender.service.SendKinServiceBase
+import org.kinecosystem.transfer.sender.view.TransferBarView
+import org.kinecosystem.transfer.sender.view.TransferInfo
+
+class TransferAmountPresenter(receiverAppName: String, private val senderAppName: String, private val receiverPublicAddress: String, private val repository: EcosystemAppsRepository, private val senderServiceBinder: SenderServiceBinder) : BasePresenter<ITransferAmountView>(), ITransferAmountPresenter, SenderServiceBinder.BinderListener {
```

Comment:
Comments to reduce dependencies / make this reusable.
Can we remove the dependency of EcosystemAppsRepository and SenderServiceBinder. 
You can do this by:
a. Move out the functionality that actually handles the transfer into a separate component and provide a listener to the presenter. 
b. provide the appIconUrl as a parameter instead of getting it from the app 
c. provide an interfaces for requesting balance and receiving result (similar to the binder but more generic) 
The component that takes care of the requesting balance and receiving result can also handle the caching 

Also if the receiverAppName/senderAppName/receiverPublicAddress are not really presented in the Calculator, but only used for the transfer, then these can be moved out of here too.